### ISSUE TITLE
for one image，regardless of how the alpha channel is modified，feature similarity is consistently above 0.97  （even between mask=0 and mask=1)

### ISSUE THREAD

**Issue by llf1234** [SunzeY/AlphaCLIP#12]

**Comment by llf1234** [comment]
test code is like:
image = preprocess(raw_image).unsqueeze(0).to("cuda:0")
    w,h = raw_image.size
    for bbox in record[-1]:
        cls,x1,y1,x2,y2,score = bbox
        x1 = int(x1)
        y1 = int(y1)
        x2 = int(x2)
        y2 = int(y2)
        binary_mask = np.zeros([h,w])
        binary_mask[y1:y2,x1:x2]=1
        alpha = mask_transform((binary_mask*255).astype(np.uint8))
        alpha = alpha.cuda().unsqueeze(dim=0).to("cuda:0")
        with torch.no_grad():
            image_features = model.visual(image, alpha)
            image_features = image_features / image_features.norm(dim=1, keepdim=True)

            binary_mask = np.ones([h,w]) ## binary_mask = np.zeros([h,w]) 
            alpha = mask_transform((binary_mask*255).astype(np.uint8))
            alpha = alpha.cuda().unsqueeze(dim=0).to("cuda:0")

            ori_features = model.visual(image,alpha)
            ori_features = ori_features / ori_features.norm(dim=1, keepdim=True)
            print(np.dot(image_features.cpu().numpy(),ori_features.cpu().numpy().T))

**Comment by Aleafy** [comment]
Please use image and text feature for dot product. For an example use, please refer to https://github.com/SunzeY/AlphaCLIP/blob/main/README.md#how-to-use.
